src/NetAnalyzers/CSharp/Microsoft.CodeQuality.Analyzers/QualityGuidelines/CSharpAssigningSymbolAndItsMemberInSameStatement.Fixer.cs

```diff
@@ -0,0 +1,123 @@
+                trailingTrivia = trailingTrivia.Where(t => t.IsKind(SyntaxKind.WhitespaceTrivia) || t.IsKind(SyntaxKind.EndOfLineTrivia)).ToSyntaxTriviaList();
+            }
+
+            var title = MicrosoftCodeQualityAnalyzersResources.AssigningSymbolAndItsMemberInSameStatementTitle;
+
+            var replacements1 = replacements.Concat(GetAssignmentExpressionStatement(assignment.Left, mostRightMember, leadingTrivia, trailingTrivia));
+            var replacements2 = replacements.Concat(GetAssignmentExpressionStatement(members, leadingTrivia, trailingTrivia));
+
+#pragma warning disable RS1010 // Provide an explicit value for EquivalenceKey - false positve 
+            var nestedCodeAction = CodeAction.Create(title, ImmutableArray.Create<CodeAction>(
+                new MyCodeAction($"{title} 1", ct => GetDocument(context.Document, root, assignment.Parent, replacements1)),
+                new MyCodeAction($"{title} 2", ct => GetDocument(context.Document, root, assignment.Parent, replacements2))
+                ), isInlinable: false);
+#pragma warning restore RS1010
+
+            context.RegisterCodeFix(nestedCodeAction, context.Diagnostics);
+        }
+
+        /// <summary>
+        /// If the assignment expression is:  a = b = c = d
+        /// Return a stack containing a, b, c, d with `a` at the bottom and `d` at the top.
+        /// </summary>
+        private static Stack<ExpressionSyntax> GetAssignmentMembers(AssignmentExpressionSyntax node)
+        {
+            var stack = new Stack<ExpressionSyntax>();
+            ExpressionSyntax current = node;
+            while (current is AssignmentExpressionSyntax assignment)
+            {
+                stack.Push(assignment.Left);
+                current = assignment.Right;
+            }
+            stack.Push(current);
+            return stack;
+        }
+
+        private static ExpressionStatementSyntax GetAssignmentExpressionStatement(Stack<ExpressionSyntax> stack, SyntaxTriviaList leadingTrivia, SyntaxTriviaList trailingTrivia)
+        {
+            var right = stack.Pop();
+            var left = stack.Peek();
+            return GetAssignmentExpressionStatement(left, right, leadingTrivia, trailingTrivia);
+        }
+
+        private static ExpressionStatementSyntax GetAssignmentExpressionStatement(ExpressionSyntax left, ExpressionSyntax right, SyntaxTriviaList leadingTrivia, SyntaxTriviaList trailingTrivia)
+            => SyntaxFactory.ExpressionStatement(
+                SyntaxFactory.AssignmentExpression(SyntaxKind.SimpleAssignmentExpression, left, right))
+                    .WithLeadingTrivia(leadingTrivia).WithTrailingTrivia(trailingTrivia);
+
+        private static Task<Document> GetDocument(Document document, SyntaxNode root, SyntaxNode oldNode, IEnumerable<SyntaxNode> replacements)
+            => Task.FromResult(document.WithSyntaxRoot(root.ReplaceNode(oldNode, replacements)));
```

Comment:
I am not sure but maybe using the `DocumentEditor` would help with solving the issue you are facing.

Reply:
@Evangelink Can you elaborate more on how I should use `DocumentEditor` to fix this?